src/id3/v2/frame/id.rs

```diff
@@ -1,45 +1,47 @@
+use std::borrow::Cow;
+
 use crate::error::{ID3v2Error, ID3v2ErrorKind, LoftyError, Result};
 use crate::tag::item::ItemKey;
 use crate::tag::TagType;
 
 /// An `ID3v2` frame ID
 #[derive(PartialEq, Clone, Debug, Eq, Hash)]
-pub enum FrameID {
+pub enum FrameID<'a> {
 	/// A valid `ID3v2.3/4` frame
-	Valid(String),
+	Valid(Cow<'a, str>),
 	/// When an `ID3v2.2` key couldn't be upgraded
 	///
 	/// This **will not** be written. It is up to the user to upgrade and store the key as [`Id3v2Frame::Valid`](Self::Valid).
 	///
 	/// The entire frame is stored as [`ItemValue::Binary`](crate::ItemValue::Binary).
-	Outdated(String),
+	Outdated(Cow<'a, str>),
 }
 
-impl FrameID {
+impl<'a> FrameID<'a> {
 	/// Attempts to create a `FrameID` from an ID string
 	///
 	/// # Errors
 	///
 	/// * `id` contains invalid characters (must be 'A'..='Z' and '0'..='9')
 	/// * `id` is an invalid length (must be 3 or 4)
-	pub fn new(id: &str) -> Result<Self> {
-		Self::verify_id(id)?;
+	pub fn new(id: Cow<'a, str>) -> Result<Self> {
```

Comment:
```suggestion
	pub fn new<I>(id: I) -> Result<Self>
	where
		I: Into<Cow<'a, str>>,
	{
```

I think this should be generic, I'd rather callers not have to deal with `Cow`s outside of return types.

Reply:
Agreed. I found a way to avoid code bloat by monomorphization.